headers/validation.h

```diff
@@ -222,14 +283,25 @@ struct NormalDistributionGenerator {
 
 	const T mean_, stddev_;
 
-	explicit NormalDistributionGenerator(T mean, T stddev) : mean_(mean), stddev_(stddev) {}
+	explicit NormalDistributionGenerator(T mean, T stddev) : mean_(mean), stddev_(stddev) {
+		assert(stddev_ >= 0);
+	}
 
 	// NOTE: Currently this retries instead of clamping to the interval.
 	T operator()(T low, T high, std::mt19937_64& rng) const {
+		assert(low < high);
 		T v;
-		do {
-			v = std::normal_distribution<T>(mean_, stddev_)(rng);
-		} while(v < low or v > high);
+		while (true) {
+			T u1 = Random::real64(rng);
+			T u2 = Random::real64(rng);
+			// Box-Muller-Methode
+			v = std::sqrt(-2.0l * std::log(u1)) * std::cos(2.0l * C20::PI * u2);
+			v = std::sqrt(stddev_)*v+mean_;
+			if (v >= low && v < high) return v;
+			v = std::sqrt(-2.0l * std::log(u1)) * std::sin(2.0l * C20::PI * u2);
```

Comment:
Hmm, are `cos` and `sin` independent enough here? If the interval is small, the fact that `cos(u2)` is outside the interval may bias the cases where `sin(u2)` is inside

Reply:
Wikipedia says this is fine